railties/lib/rails/generators/rails/app/app_generator.rb

```diff
@@ -114,7 +114,7 @@ def config
 
         directory "environments"
         directory "initializers"
-        directory "locales"
+        directory "locales" unless options[:update]
```

Comment:
Since this needs to be fixed manually when `locals` changes, there are maintainability problems still. 

We need to be able to do this check dynamically. 
(I also think that the approach of calculating differences with `rake release` is appropriate)

Reply:
`directory "locales"` just copy the directory existing in [template folder](https://github.com/rails/rails/tree/master/railties/lib/rails/generators/rails/app/templates).
Like environments and initializers directories specified before the line.
Please tell me a little more about the maintainability problems you mentioned.

Reply:
If more locale files will be added in future rails versions, we will need to implement some codes in `config_when_updating` method like recently added files under the `config/initializers` directory. I think it's not a problem.